contracts/contracts/harvest/BaseHarvester.sol

```diff
@@ -58,38 +84,46 @@ abstract contract BaseHarvester is Governable {
     /**
      * Address receiving rewards proceeds. Initially the Vault contract later will possibly
      * be replaced by another contract that eases out rewards distribution.
-     */
+     **/
     address public rewardProceedsAddress;
 
     /**
-     * @dev Constructor to set up initial internal state
-     * @param _vaultAddress Address of the Vault
-     */
-    constructor(address _vaultAddress) {
-        require(address(_vaultAddress) != address(0));
+     * All tokens are swapped to this token before it gets transferred
+     * to the `rewardProceedsAddress`. USDT for OUSD and WETH for OETH.
+     **/
+    address public immutable baseTokenAddress;
+    // Cached decimals for `baseTokenAddress`
+    uint256 public immutable baseTokenDecimals;
+
+    // Uniswap V2 path for reward tokens using Uniswap V2 Router
+    mapping(address => address[]) public uniswapV2Path;
+    // Uniswap V3 path for reward tokens using Uniswap V3 Router
+    mapping(address => bytes) public uniswapV3Path;
+    // Pool ID to use for reward tokens on Balancer
+    mapping(address => bytes32) public balancerPoolId;
+    // Packed indices of assets on the Curve pool
```

Comment:
mental note: we should have a test that confirms these 4 mappings are correctly configured. Not only by checking values in fork tests but also fork tests doing queries to e.g. Curve and check that `curvePoolIndices` return correct values for queried assets. 

In short if we only change the reward token address to use a different Curve pool to swap out the reward token,(and forget to update `curvePoolindices`) a test should fail.

Reply:
nvm, you do check for this one

Reply:
Yeah, the check is there for Curve. For Uniswap V2 and V3, I just parse the path and make sure the source and destination token are as expected. For Balancer, I couldn't do much there, other than null check. I could probably query the pool and check if the pool tokens match. Will make that change

Reply:
Yup that sounds like a great suggestion you could call `getPoolTokens` on the Balancer Vault with the given poolId to see if all the tokens there are expected.

I think that ETH (that can be converted to WETH) will be shown as a `0xEeeeEe....`